main.go

```
@@ -0,0 +1,107 @@
+		"localhost:8000",
+		grpc.WithInsecure(),
+		grpc.FailOnNonTempDialError(true),
+		grpc.WithBlock(),
+	)
+	if err != nil {
+		log.Fatalln("error: ", err)
+	}
+	defer conn.Close()
+
+	client := bpb.NewBirthdayFunctionsClient(conn)
+
+	return client
+}
+
+//function that inserts a birthday object
+func (r *Router) CreateBirthday(c *gin.Context) {
+	name := c.Request.FormValue("name")
+	personalNumber := c.Request.FormValue("personalNumber")
+	date := c.Request.FormValue("date")
+
+	request := &bpb.CreateBirthdayRequest{
+		PersonalNumber: personalNumber, Name: name, Date: date}
+	res, err := r.client.CreateBirthday(c, request)
+	if err != nil {
+		fmt.Println("create birthday method failed. error: ", err)
+	}
+
+	c.JSON(200, res)
+}
+
+//function that read/get all birthday objects
+func (r *Router) GetBirthday(c *gin.Context) {
+	personalNumber := c.Query("personalNumber")
+
+	request := &bpb.GetBirthdayRequest{PersonalNumber: personalNumber}
+	res, err := r.client.GetBirthday(c, request)
+	if err != nil {
+		fmt.Println("get birthday method failed. error: ", err)
+	}
+
+	c.JSON(200, res)
+}
+
+//function that updates a birthday object by personalNumber
+func (r *Router) UpdateBirthday(c *gin.Context) {
+	personalNumber := c.Request.FormValue("personalNumber")
+	name := c.Request.FormValue("name")
+	date := c.Request.FormValue("date")
+
+	request := &bpb.UpdateBirthdayRequest{PersonalNumber: personalNumber, Name: name, Date: date}
+	res, err := r.client.UpdateBirthday(c, request)
+	if err != nil {
+		fmt.Println("update birthday method failed. error: ", err)
+	}
+	c.JSON(200, res)
+}
+
+//function that deletes a certain birthday object by personal number
+func (r *Router) DeleteBirthday(c *gin.Context) {
+	personalNumber := c.Query("personalNumber")
+
+	request := &bpb.DeleteBirthdayRequest{PersonalNumber: personalNumber}
+	res, err := r.client.DeleteBirthday(c, request)
+	if err != nil {
+		fmt.Println("delete birthday method failed. error: ", err)
```

so after the error is printed, the result is success? (200)
you should return an error message and an error code to display to the client

---

done